src/main/java/sirius/db/mixing/types/MultiLanguageString.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.db.mixing.types;
+
+import sirius.kernel.Sirius;
+import sirius.kernel.health.Exceptions;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Provides a language-text map as property value.
+ * <p>
+ * These are stored in MongoDB as an array containing sub-documents
+ * containing a <tt>lang</tt> and <tt>text</tt> properties
+ *
+ * @see sirius.db.mixing.properties.MultiLanguageStringProperty
+ */
+public class MultiLanguageString extends SafeMap<String, String> {
+
+    private static final List<String> supportedLanguages =
+            Sirius.getSettings().getConfig("mongo").getStringList("supportedLanguages");
+
+    private String defaultLanguage;
+
+    /**
+     * Creates a new multi-language string property without default language.
+     */
+    public MultiLanguageString() {
+        this(null);
+    }
+
+    /**
+     * Creates a new multi-language string property with default language.
+     *
+     * @param defaultLanguage the default language code
+     */
+    public MultiLanguageString(String defaultLanguage) {
+        this.defaultLanguage = defaultLanguage;
+    }
+
+    @Override
+    protected boolean valueNeedsCopy() {
+        return false;
+    }
+
+    @Override
+    protected String copyValue(String value) {
+        return value;
+    }
+
+    /**
+     * Adds a new text for the given language.
+     *
+     * @param language the language code
+     * @param text     the text associated with the language
+     * @return the object itself for fluent method calls
+     */
+    public MultiLanguageString addText(String language, String text) {
+        assertLanguage(language);
+        put(language, text);
+        return this;
+    }
+
+    /**
+     * Checks if a text exists for a given language.
+     *
+     * @param language the language code
+     * @return <tt>true</tt> when a text exists, otherwise <tt>false</tt>
+     */
+    public boolean hasText(String language) {
+        return data().containsKey(language);
+    }
+
+    /**
+     * Returns the text associated with the default language.
+     *
+     * @return the text
+     */
```

Comment:
@throws HandledException if no transaltion is available for the <tt>defaultLanguage</tt>